pkg/api/unversioned/types.go

```diff
@@ -339,6 +339,8 @@ type APIResource struct {
 	Name string `json:"name"`
 	// namespaced indicates if a resource is namespaced or not.
 	Namespaced bool `json:"namespaced"`
+	// kind is the kind for the resource (e.g. 'Foo' is the kind for a resource 'foo')
+	Kind string `json:"kind"`
```

Comment:
`kind` is definitely needed in order to support non-hardcoded discovery from kubectl. 

We'll also need a list of subresources, such as scale, supported by the resource, but that doesn't need to be added by this PR.

cc @lavalamp